core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala

```diff
@@ -339,7 +339,8 @@ class HadoopMapReduceWriteConfigUtil[K, V: ClassTag](conf: SerializableConfigura
 
   override def createCommitter(jobId: Int): HadoopMapReduceCommitProtocol = {
     FileCommitProtocol.instantiate(
-      className = classOf[HadoopMapReduceCommitProtocol].getName,
+      className = getConf.get("mapreduce.sources.commitProtocolClass",
```

Comment:
One workaround might be:

```scala
    val config = new HadoopMapReduceWriteConfigUtil[K, V](new SerializableConfiguration(conf)) {
      override def createCommitter(jobId: Int): HadoopMapReduceCommitProtocol = {
        FileCommitProtocol.instantiate(
          className = "CustomClassName",
          jobId = jobId.toString,
          outputPath = new SerializableConfiguration(
            conf).value.get("mapreduce.output.fileoutputformat.outputdir")
        ).asInstanceOf[HadoopMapReduceCommitProtocol]
      }
    }
    SparkHadoopWriter.write(
      rdd = self,
      config = config)
```

Although `HadoopMapReduceWriteConfigUtil` would have to be inherited within `org.apache.spark` package in your project though (if you're using Scala). For Java, it won't validate the package name.